static.config.js

```diff
@@ -21,16 +21,19 @@ const makeAllPages = async (langCode) => {
   const path = `/${!!langCode ? langCode : ''}`
   console.log(`- Building routes for ${path}...`);
 
+  const client = createGraphQLClientsByLang(langCode);
+  const { allThemes } = await client.request(allThemesQuery);
+
   const data = {
     path: path,
     component: 'src/js/pages/Home',
-    children: await makeChildPages(langCode),
+    children: await makeChildPages(client, allThemes),
```

Comment:
Simi is right... always. I think our last convo about this may have led you down this path but this feels like the perfect use case for `allSiteData`.

Reply:
I looked at `allSiteData` initially, but had trouble figuring out how we'd make that work with setting the language code. My thought was that we'd have to generate all language data and store it with language code as the key (see below). Then we'd have to set `languageCode` as a property and have each component get the data by that code. That seemed like pushing too much logic into components and still required us to send `languageCode` to each component, but maybe there's a better way to do it.

```
allSiteData: {
  en: {
    allThemes
  },
  es: {
    allThemesES
  },
  ...
}
```

Reply:
I'm not sure I understand. Would Menu.js be the only component that needed to know language code? No less, we can inject language code as needed to components via wrapping the component in react-intl's HOC method injectIntl -- so like
```
const Submenu = ({..., intl}) => {
    let appLanguageCode = intl.locale; //<--this is the language code
    ...
}
export default injectIntl(Submenu);
```

Reply:
Looks like they _just_ changed their docs. Curiously, `getSiteData` doesn't show up anymore :|

Good point, only Menu.js would need to use `allSiteData`. It would then need to use that language code to get the `allSiteData` that corresponds with the language it's trying to render.

```
const Menu = ({..., intl}) => {
    let appLanguageCode = intl.locale; //<--this is the language code
    let allThemes = navigationData[appLanguageCode]; // navigationData is passed in via allSiteData
}
export default injectIntl(Menu);
```

Where `getSiteData` would need to fetch all data for all languages (not just the language we're currently building)

```
getSiteData() = async () => ({
  en: createGraphQLClientsByLang('en').request(allThemesQuery),
  es: createGraphQLClientsByLang('es').request(allThemesQuery),
  vi: createGraphQLClientsByLang('vi').request(allThemesQuery),
  ar: createGraphQLClientsByLang('ar').request(allThemesQuery),
})
```

### TL;DR
Both solutions seem a little weird to me. Let me try out the `allSiteData` version and see how it nets out.

Reply:
TRUTH. Thanks for the explanation and agreed. It's just going to get more complicated. And we haven't even added Esperanto.

Reply:
Another check in the "Simi is always right" column https://github.com/cityofaustin/janis/pull/211